### DISCUSSION TITLE
Pressure measurement

### DISCUSSION THREAD

**Discussion by mp-se** [thorrak/fermentrack#587] [Q&A] [answered]
I've been starting with doing some pressure fermentation and I would be nice to get the pressure data into fermentrack, I belive it should be possible to extend the beerpi firmware to collect that data.

However I've been trying to find some good sensors that could interpret the signal (either via brewpi) or some new controller built on esp8266/esp32. 

Have anyone found any good sensors to use for measuring pressure?

**Comment by lbussy** [comment, marked as answer]
The one @DJMarlow uses [here](https://www.homebrewtalk.com/threads/flite-keg-level-temperature-and-pressure-sensor.687681/) might work.

**Reply by thorrak** [reply]
Honestly, I hadn't thought about it, but there's nothing preventing me from getting Flite support added to Fermentrack...

**Reply by DJMarlow** [reply]
I guess we'll have to add it to the list!

**Comment by mp-se** [comment]
That a good suggestion, I will take a closer look if I can adapt that one. 

I'm using a fermzilla with spunding valve to control my fermentation and I want to replace the pressure gauge with a sensor instead (or complement, like in the picture). So if this could be adapted to fit a ball-lock post instead it can be used for any vessel. 

![image](https://user-images.githubusercontent.com/51722613/105078831-d5937a00-5a8e-11eb-9426-d273b8e79bee.png)

**Comment by mp-se** [comment]
The sensor used is using I2C, could we not just hook this up to the brewpi controller and collect the data that way? It would probably be possible to 3d print a adapter to fit in the JB/Kegland fittings...

**Reply by thorrak** [reply]
> The sensor used is using I2C, could we not just hook this up to the brewpi controller and collect the data that way? It would probably be possible to 3d print a adapter to fit in the JB/Kegland fittings...

Could it be done? Sure. I would 100% recommend against it though. ;)

ESPs are cheap, and having firmware designed to do exactly what you want is better than having to run an additional cable between your temperature controller and your pressure sensor, running the risk of the pressure code causing crashes/issues with the temperature control, etc.

**Reply by lbussy** [reply]
> ESPs are cheap, and having firmware designed to do exactly what you want is better than having to run an additional cable between your temperature controller and your pressure sensor, running the risk of the pressure code causing crashes/issues with the temperature control, etc.

^^^^ THAT

People forget what the controller is supposed to be doing I think. :)

**Reply by mp-se** [reply]
Well hooking up a esp directly to the senor that can push senor values could be an approach then. Similar to the iSpindel would do probably work fine. I think I will give that a try, would not be that difficult to put together. I probably have most of the parts lying around at home anyway.

**Reply by mp-se** [reply]
I guess the ESP32 port that you have been working on @thorrak would be up to the job to handle some extra I2C communication. i would expect the display updates over i2c to be using more compute power than reading a sensor....

**Comment by DJMarlow** [comment]
The pressure sensor I spec'd for the Flite sensor is HSCMAND150PA4A3. Here's the datasheet on this sensor line:
https://sensing.honeywell.com/index.php?ci_id=151133

There are a few considerations here:

1. Range - This sensor is a 0-150 PSI range. If you are measuring a lower range, consider getting a lower range sensor if this accuracy does not meet your needs. However, I have found this range seems to be accurate within 1 PSI. If you get a lower range, you will need to modify the scaling in the code.
2. Media - This sensor is designed for gas media, not liquid. Make sure the pressure transducer is not installed in a way that it is measuring a pressurized liquid media
3. Interface - This particular sensor is an I2C device with an address of 0x48. You can connect this sensor to an I2C bus and read the data directly. You can use the Flite python lib and RPi support here:
https://www.flitesense.com/raspberry-pi-integration

The Flite lib is designed for connecting an entire Flite sensor, but it would technically still work if you wanted to only get pressure/temperature from the HSC sensor. Make sure you instantiate a Flite object as BLACK if you buy the 0x48 sensor.

Here's Honeywell's literature I used to write the Flite lib:
https://sensing.honeywell.com/i2c-comms-digital-output-pressure-sensors-tn-008201-3-en-final-30may12.pdf

**Reply by mp-se** [reply]
Thanks for the summary. Your project looks very nice as way to keep track of what's left in the kegs. Will definitely look into that some more. 

I was researching the sensor that you used for the project and for measuring fermentation it's would suffice with a senor that can handle up to 50 psi. No fermentation vessel would handle more than 2.5 bar anyway. I saw that they had a few versions that could handle less pressure that are a little cheaper.  

I would probably try to hook it up to some kind of Arduino board with WIFI just to make it easy to move around, I already have a basic project that does the wifi, rest api's etc so I would not be that difficult. 

I was not planning to submerge it into the liquid but instead connect it before my spunding valve (replace the manometer in that picuture) so it would measure the Co2 pressure in the vessel. 

My main problem will probably be to get my hands on one of these, I havent found any shops that sell them in Sweden yet so I will probably need to look to import it from somewhere.

**Reply by DJMarlow** [reply]
Here's where I get them. Not cheap, but they work well:
https://www.digikey.com/en/products/detail/honeywell-sensing-and-productivity-solutions/HSCMAND150PA4A3/2416036

If you are planning on wired, you can find many analog sensors for pressure only for a fraction of this.

**Reply by mp-se** [reply]
I like the digital stuff better, but I will look around. 

I checked the site where you bought yours and if I buy them it's around 60-70$ for one sensor including freight and import taxes... Not really what I want to pay for a component. 

Since I will be measuring between 0-35 psi I want something with good precision so I guess they will cost. 

I have found the version with standard precision but not the high precision that you used for around 25-40$ locally. have you tried that version to see what the precision is?

**Comment by DJMarlow** [comment]
Or @thorrak can add Flite support to Fermentrack and you can use the Flite controller's ESP8266 wifi to communicate the pressure data. No cables needed, DIY and firmware already developed and tested...

**Reply by thorrak** [reply]
I keep forgetting - fermenters and kegs aren’t the same thing. 

Guess what I used for fermenters for years. 😂

**Reply by mp-se** [reply]
I could probably help with that as well. First thing is to get the parts and build it then. 

I guess I could design a 3d fitting that I could connect to the gas output instead of the keg lid would be a faster way forward in that case...

**Comment by mp-se** [comment]
Well today I finally got the pressure sensors. After some testing with analog sensors i quickly skipped that path and went with the Honeywell sensors. I have a running prototype on a esp8266 and it seams to work fine... Now I need to fix some kind of enclosure and secure that it can handle the pressure during fermentation.
